evalml/pipelines/component_graph.py

```diff
@@ -189,7 +200,7 @@ def _compute_features(self, component_list, X, y=None, fit=False):
             else:
                 if fit:
                     component_instance.fit(input_x, input_y)
-                if not (fit and component_instance.name == self.get_last_component().name):  # Don't call predict on the final component during fit
+                if not (fit and component_name == self.compute_order[-1]):  # Don't call predict on the final component during fit
```

Comment:
How come we're not using `get_last_component().name`?

Reply:
`get_last_component().name` will use the generic name of the component, rather than the custom name given to that specific component. If there are multiples of the same estimator in the pipeline and one of them is the final component, it will not call predict on the non-final one.

Reply:
Thanks for explaining!